File: Java/_447_Number_of_Boomerangs/Solution.java
class Solution {
    public int numberOfBoomerangs(int[][] points) {
        int res = 0;
        for (int i = 0; i < points.length; i++)
        {
            Map<Integer, Integer> map = new HashMap<Integer, Integer>();
            for (int j = 0; j < points.length; j++)
            {
                if (map.containsKey(distance(points[i], points[j]))) {
                    map.put(distance(points[i], points[j]), map.get(distance(points[i], points[j])) + 1);
                }
                else {
                    map.put(distance(points[i], points[j]), 1);
                }
            }
            // for (Map.Entry<Integer, Integer> entry : map.entrySet()) {  
            //     res += entry.getValue() * (entry.getValue() - 1);  
            // }
            for (int dis : map.values())
                res += dis * (dis - 1);
        }
        return res;
    }
    
    public int distance(int[] px, int[] py) {
        return (px[0] - py[0]) * (px[0] - py[0]) + (px[1] - py[1]) * (px[1] - py[1]);
    }
}

File: Python/_447_Number_of_Boomerangs/solution.py
class Solution:
    def numberOfBoomerangs(self, points):
        """
        :type points: List[List[int]]
        :rtype: int
        """
        count = 0
        for i in range(len(points)):
            maps = {}
            for j in range(len(points)):
                distance = pow(points[i][0] - points[j][0], 2) + pow(points[i][1] - points[j][1], 2)
                maps[distance] = maps.get(distance, 0) + 1
            for v in maps.values():
                count += v * (v - 1)
        return count

